src/catalog_provider/glue.rs

```diff
@@ -29,460 +25,127 @@ pub enum TableRegistrationOptions {
-        };
-
-        Ok(listing_options)
-    }
-
-    fn map_glue_data_type_to_arrow_data_type(glue_data_type: &GlueDataType) -> Result<DataType> {
-        match glue_data_type {
-            GlueDataType::TinyInt => Ok(DataType::Int8),
-            GlueDataType::SmallInt => Ok(DataType::Int16),
-            GlueDataType::Int => Ok(DataType::Int32),
-            GlueDataType::Boolean => Ok(DataType::Boolean),
-            GlueDataType::BigInt => Ok(DataType::Int64),
-            GlueDataType::Float => Ok(DataType::Float32),
-            GlueDataType::Double => Ok(DataType::Float64),
-            GlueDataType::Binary => Ok(DataType::Binary),
-            GlueDataType::Timestamp => Ok(DataType::Timestamp(TimeUnit::Nanosecond, None)),
-            GlueDataType::String => Ok(DataType::Utf8),
-            GlueDataType::Char => Ok(DataType::Utf8),
-            GlueDataType::Varchar => Ok(DataType::Utf8),
-            GlueDataType::Date => Ok(DataType::Date32),
-            GlueDataType::Decimal(precision, scale) => {
-                Ok(DataType::Decimal256(*precision as u8, *scale as u8))
-            }
-            GlueDataType::Array(inner_data_type) => {
-                let array_arrow_data_type =
-                    Self::map_glue_data_type_to_arrow_data_type(inner_data_type)?;
-                Ok(DataType::List(Box::new(Field::new(
-                    "item",
-                    array_arrow_data_type,
-                    true,
-                ))))
-            }
-            GlueDataType::Map(key_glue_data_type, value_glue_data_type) => {
-                let key_arrow_data_type =
-                    Self::map_glue_data_type_to_arrow_data_type(key_glue_data_type)?;
-                let value_arrow_data_type =
-                    Self::map_glue_data_type_to_arrow_data_type(value_glue_data_type)?;
-                Ok(DataType::Map(
-                    Box::new(Field::new(
-                        "key_value",
-                        DataType::Struct(vec![
-                            Field::new("key", key_arrow_data_type, true),
-                            Field::new("value", value_arrow_data_type, true),
-                        ]),
-                        true,
-                    )),
-                    true,
-                ))
-            }
-            GlueDataType::Struct(glue_fields) => {
-                let mut fields = Vec::new();
-                for glue_field in glue_fields {
-                    let field_arrow_data_type =
-                        Self::map_glue_data_type_to_arrow_data_type(&glue_field.data_type)?;
-                    fields.push(Field::new(&glue_field.name, field_arrow_data_type, true));
-                }
-                Ok(DataType::Struct(fields))
-            }
+impl GlueSchemaProvider {
+    async fn create_table(&self, glue_table: &Table) -> Result<Arc<dyn TableProvider>> {
+        log::debug!("glue table: {:#?}", glue_table);
+
+        let provider = glue_table
```

Comment:
In case the table was not created/registered with spark but with (eg: Athena) this field will not exist.
In those cases there is a "table_type" table property which has value delta (or iceberg).

Reply:
Thanks, good point, taking a look on it.

Reply:
Done